File: mobile-edge/clients/typescript/edgeClient.ts

export class EdgeClient {
  constructor(private base:string, private bearer?:string){}
  private async req(path:string, opt:RequestInit={}){
    const url = this.base.replace(/\/$/,'') + path;
    opt.headers = Object.assign({"Content-Type":"application/json"}, opt.headers||{});
    if (this.bearer) (opt.headers as any)["Authorization"] = this.bearer;
    const r = await fetch(url, opt);
    if (!r.ok) throw new Error(`HTTP ${r.status}`);
    const txt = await r.text();
    try { return JSON.parse(txt); } catch { return txt; }
  }
  config(){ return this.req("/mobile/config"); }
  entitlement(){ return this.req("/mobile/entitlement"); }
  pushRegister(deviceId:string, token:string, platform?:string){
    return this.req("/mobile/push/register", { method:"POST", body: JSON.stringify({ deviceId, token, platform }) });
  }
  receiptVerify(idemKey:string, body:{transactionId:string,productId:string,platform:"android"|"ios",receipt:string}){
    return this.req("/mobile/receipt/verify", { method:"POST", body: JSON.stringify(body), headers: {"Idempotency-Key": idemKey } });
  }
  analyticBatch(signature:string, events:any[]){
    const body = JSON.stringify({ events });
    return this.req("/mobile/analytic/batch", { method:"POST", body, headers: {"X-SR-Analytic-Signature": signature} });
  }
}

File: mobile-edge/clients/kotlin/EdgeClient.kt

package mobile.edge.client
import java.net.HttpURLConnection
import java.net.URL

class EdgeClient(private val base:String, private val bearer:String? = null){
    private fun req(path:String, method:String="GET", body:String?=null, headers: Map<String,String> = emptyMap()): Pair<Int,String>{
        val url = URL(base.trimEnd('/') + path)
        val c = (url.openConnection() as HttpURLConnection)
        c.requestMethod = method
        for ((k,v) in headers) c.setRequestProperty(k, v)
        if (bearer != null) c.setRequestProperty("Authorization", bearer)
        if (body != null){
            c.doOutput = true
            c.setRequestProperty("Content-Type", "application/json")
            c.outputStream.use { it.write(body.toByteArray()) }
        }
        val code = c.responseCode
        val stream = if (code in 200..299) c.inputStream else c.errorStream
        val txt = stream?.readBytes()?.toString(Charsets.UTF_8) ?: ""
        c.disconnect()
        return code to txt
    }
    fun config(): Pair<Int,String> = req("/config")
    fun entitlement(): Pair<Int,String> = req("/entitlement")
    fun pushRegister(deviceId:String, token:String, platform:String?=null): Pair<Int,String> {
        val b = "{"deviceId":"$deviceId","token":"$token","platform":"${platform ?: ""}"}"
        return req("/push/register", "POST", b)
    }
    fun receiptVerify(idem:String, transactionId:String, productId:String, platform:String, receipt:String): Pair<Int,String> {
        val b = "{"transactionId":"$transactionId","productId":"$productId","platform":"$platform","receipt":"$receipt"}"
        return req("/receipt/verify", "POST", b, headers = mapOf("Idempotency-Key" to idem))
    }
}
